unlock-app/src/components/interface/Header.js

```diff
@@ -17,18 +17,63 @@ const navigationButtons = [
   Buttons.Telegram,
 ]
 
+const accountTypes = {
+  managed: 'account/managed',
+  crypto: 'account/crypto',
+}
+
+export const typeOfAccount = account => {
+  if (!account) {
+    // this case mostly seems to happen in component tests that include the
+    // header but haven't had an account set.
+    return 'account/undefined'
+  }
+  const { address, emailAddress } = account
+  if (address && emailAddress) {
+    return accountTypes.managed
+  } else if (address) {
+    return accountTypes.crypto
+  } else {
+    // This final catch-all state is invalid. No app navigation will display for
+    // a user who doesn't match the above conditions. A user may be in an
+    // intermediate state (not all account updates have hit redux), which may
+    // cause a brief flicker of the icons. The return value here should indicate
+    // what pieces are missing for debugging purposes.
+    return `account/address={${address}}--emailAddress={${emailAddress}}`
+  }
+}
+
 // distinct from the above buttons are page nav buttons -- they are only visible
 // from within the app and not from the homepage or other static pages.
 const appButtons = [
-  { Button: PageNavButtons.Dashboard, page: '/dashboard' },
-  { Button: PageNavButtons.Log, page: '/log' },
+  {
+    Button: PageNavButtons.Dashboard,
+    page: '/dashboard',
+    allowedUsers: [accountTypes.crypto],
+  },
+  {
+    Button: PageNavButtons.Log,
+    page: '/log',
+    allowedUsers: [accountTypes.crypto, accountTypes.managed],
+  },
+  {
+    Button: PageNavButtons.Settings,
+    page: '/settings',
+    allowedUsers: [accountTypes.managed],
+  },
+  {
+    Button: PageNavButtons.KeyChain,
+    page: '/keychain',
+    allowedUsers: [accountTypes.crypto, accountTypes.managed],
```

Comment:
Really appreciate the explicit nature of allowedUsers here so we don't have to do crazy tracing to figure out how people ended up in given paths! 👍